opentelemetry-sdk/src/trace/span_processor.rs

```diff
@@ -460,29 +459,37 @@ impl BatchSpanProcessor {
     where
         E: SpanExporter + Send + Sync + 'static,
     {
-        // Get upto `max_export_batch_size` amount of spans from the channel and push them to the span vec
-        while let Ok(span) = spans_receiver.try_recv() {
-            spans.push(span);
-            if spans.len() == config.max_export_batch_size {
-                break;
+        let target = current_batch_size.load(Ordering::Relaxed); // `target` is used to determine the stopping criteria for exporting spans.
+        let mut result = OTelSdkResult::Ok(());
+        let mut total_exported_spans: usize = 0;
+
+        while target > 0 && total_exported_spans < target {
```

Comment:
`current_batch_size` might be changed by other tasks so we are using `target` which is a snapshot of  `current_batch_size` here.

If we won't run this funcion concurrently then we should consider not passing in a `AtomicUsize` as `current_batch_size`

Reply:
It can be changed by other threads, but this method only exports up to snapshot. If more items get into the queue which it is working, that'd be handled by another call to this method. IOW, this method has a target it set out to achieve - it'll work until that target is achieved and nothing more.

Also, this function is not run concurrently - but we still need `current_batch_size` as `AtomicSize` as this function updates its value based on how much it exports.